### PR TITLE
validate cluster version

### LINKED ISSUE #5450: API fields validation
SysEleven while working on terraform provider for kubermatic discovered a problem described below, info from @furkhat :

> POST Payload values validation.
>     Steps to reproduce:
>        1. Create a cluster with 'version` = '1.2.3'
>     Expected behaviour:
>           400 Bad Request
>     Actual behaviour:
>           200

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: validate cluster version during the creation

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5450

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/v1/cluster/cluster_test.go [hunk 1]
```diff
@@ -882,6 +882,16 @@ func TestCreateClusterEndpoint(t *testing.T) {
 			),
 			ExistingAPIUser: test.GenAPIUser("John", "john@acme.com"),
 		},
+		// scenario 13
```
Comment (lsviben) [inline api/pkg/handler/v1/cluster/cluster_test.go:901]: could you add a test with no version set in the body?

Comment (zreigz) [reply api/pkg/handler/v1/cluster/cluster_test.go:901]: done

**Pull request merged**